grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+import org.curioswitch.common.protobuf.json.MessageMarshaller;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.Message;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+public final class UnframedGrpcErrorHandlerBuilder {
+    private UnframedGrpcStatusMappingFunction statusMappingFunction;
+
+    private MessageMarshaller jsonMarshaller = UnframedGrpcErrorHandlers.ERROR_DETAILS_MARSHALLER;
+
+    @Nullable
+    private Set<UnframedGrpcErrorResponseType> responseTypes;
+
+    UnframedGrpcErrorHandlerBuilder(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = statusMappingFunction;
+    }
+
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        this.jsonMarshaller = requireNonNull(jsonMarshaller, "jsonMarshaller");
+        return this;
```

Comment:
Could we make `jsonMarshaller()` and `registerMarshallers()` mutually exclusive, so that the code like the following is rejected to avoid confusion:

```java
.builder()
.registerMarshallers(...)
.jsonMarshaller(...)
```